docs/mmds/mmds-design.md

```diff
@@ -41,98 +37,38 @@ desired data store structure and contents. Here's a JSON example:
 }
 ```
 
-Queries from the guest (more on them a bit later) will be applied to this
-structure. For example, a `GET` request for
-`http://169.254.169.254/latest/meta-data/ami-id` will return a response body
-consisting of *ami-12345678*. The MMDS contents can be updated either via a
-subsequent `PUT` (that replaces them entirely), or using `PATCH` requests,
-which feed the JSON body into the JSON Merge Patch functionality, based on
-[RFC 7396](https://tools.ietf.org/html/rfc7396). MMDS related API requests come
-from the host, which is considered a trusted environment, so there are no
-checks beside the kind of validation done by HTTP server and `serde-json` (the
-crate used to de/serialize JSON). Most importantly, there is currently no
-maximum size bound for MMDS contents; the users are free to provide arbitrarily
-large inputs. However, the HTTP server is likely to encounter/become a
+The MMDS contents can be updated either via a subsequent `PUT` (that replaces them
+entirely), or using `PATCH` requests, which feed the JSON body into the JSON Merge
+Patch functionality, based on [RFC 7396](https://tools.ietf.org/html/rfc7396). MMDS
+related API requests come from the host, which is considered a trusted environment,
+so there are no checks beside the kind of validation done by HTTP server and
+`serde-json` (the crate used to de/serialize JSON). Most importantly, there is
+currently no maximum size bound for MMDS contents; the users are free to provide
```

Comment:
It's also worth mentioning that https://github.com/firecracker-microvm/firecracker/issues/1336 can be hit with large payloads.